### PR TITLE
Mv 280 adding pinning

### PR DESCRIPTION
**Pull request by pkrucz00**
Draft for pinning the user

### CODE REVIEW

File: assets/src/features/room-page/utils/usePinning.tsx
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:10]: I think null would be better here instead of an empty string. It would be the correct ID or null. It would be easier to handle

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]

File: assets/src/pages/room/VideochatSection.tsx
Comment (bblaszkow06) [inline assets/src/pages/room/VideochatSection.tsx:154]: How hard would it be to handle that scenario too? I can imagine you'd like to pin all the items before someone joins, e.g. you're pinning screen share and your camera before the other peer arrives. At first I thought it was a bug that the button didn't appear

**Review by bblaszkow06 — CHANGES_REQUESTED** [review]
1. Auto pinning of screenshare is missing

Topics for design discussion
1. with 2 pinned items, and one not pinned the layout gets weird. IMO the unpinned items should be aligned to the middle of the column on right

Screenshot_20230209_182858

2. Since pinning will be handling screen share most of the time, IMO it makes more sense to use 2 rows instead of columns for two items
3. I couldn't resist trying to unpin items by clicking on a pin icon in the top right corner. It also happens to be the place where X for closing window appears and I think that's not coincident. I'd prefer that over the current unpin that covers the view

File: assets/src/pages/room/VideochatSection.tsx
Comment (pkrucz00) [reply assets/src/pages/room/VideochatSection.tsx:154]: Thank you. I'll talk to Alicja about those issues.

**Comment by bblaszkow06** [comment]
LGTM, but wait for approval from @kamil-stasiak & @alexmaar 
approval-5

**Review by bblaszkow06 — APPROVED** [review]

File: assets/src/features/room-page/components/PinComponents.tsx
Comment (alexmaar) [inline assets/src/features/room-page/components/PinComponents.tsx:36]: you can use tailwind class
```suggestion
      <Pin className="text-brand-dark-blue-500" />
```

File: assets/src/features/room-page/components/PinComponents.tsx
Comment (alexmaar) [inline assets/src/features/room-page/components/PinComponents.tsx:14]: What about using our custom `<Button/>` component?

File: assets/src/features/room-page/utils/usePinning.tsx
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:21]: You don't need to limit the history size. You could use `Set` instead of an array. 

Also `.pop()` function mutates an array and it's not permitted in `React.useState()`

File: assets/src/features/room-page/utils/usePinning.tsx
Comment (kamil-stasiak) [inline assets/src/features/room-page/utils/usePinning.tsx:16]: when you change an array to a set it will be more accurate to call it `addToHistory`

File: assets/src/pages/types.ts
Comment (alexmaar) [inline assets/src/pages/types.ts:45]: Can we avoid this repetition? You could extract common props to a separate type or create a union type for typeName: `"local" | "remote"`

File: assets/src/features/room-page/components/DisabledTrackIcon.tsx
Comment (alexmaar) [inline assets/src/features/room-page/components/DisabledTrackIcon.tsx:12]: Please use the `brand-dark-blue-500` class defined in tailwind config for `#001A72` value

File: assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx:81]: I don't see any advantages of extracting these lines to constant

File: assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx:73]: It's hard for me to reason about nested ternary operators. Could you extract it to function with well-defined simple ifs?

File: assets/src/features/room-page/utils/usePinning.tsx
Comment (alexmaar) [inline assets/src/features/room-page/utils/usePinning.tsx:10]: As it is a custom hook, I would rather move this file to `features/room-page/hooks`

File: assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx:67]: In one place you are using construction with ternary operator and null, a in another ternary operator with <>
```ts
topLeft={
  hasInitials && showDisabledIcon(pinnedTile.audio) ? (
    <DisabledMicIcon isLoading={isLoading(pinnedTile.audio)} />
  ) : null
}

...

topLeft={
  hasInitials && showDisabledIcon(config.audio) ? (
    <DisabledMicIcon isLoading={isLoading(audio)} />
  ) : (
    <></>
  )
}
```

I suggest sticking to one version:
```ts
hasInitials && showDisabledIcon(pinnedTile.audio) ? (
  <DisabledMicIcon isLoading={isLoading(pinnedTile.audio)} />
) : null
```
or use`undefined` instead of `null` and remove `| null` from Props because topLeft and others are truly optional

File: assets/src/pages/room/hooks/useMembraneClient.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/hooks/useMembraneClient.tsx:115]: I could reintroduce [object Object] error in a toast

File: assets/src/pages/room/hooks/useMembraneClient.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/hooks/useMembraneClient.tsx:140]: I could reintroduce [object Object] error in a toast

File: assets/src/pages/room/components/StreamPlayer/PeerInfoLayer.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/components/StreamPlayer/PeerInfoLayer.tsx:8]: These are optional so I think would be enough to use only `?:`

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]

File: assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx
Comment (alexmaar) [inline assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx:17]: It looks like a helper function so, please move it to `features/room-page/utils`

File: assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx
Comment (pkrucz00) [reply assets/src/pages/room/components/StreamPlayer/UnpinnedTilesSection.tsx:47]: I've left it as it is. I think it is readable enough, especially if well formatted.

File: assets/src/pages/room/hooks/usePinning.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/hooks/usePinning.tsx:16]: Please create new Set here

**Review by kamil-stasiak — APPROVED** [review]

File: assets/src/pages/room/hooks/usePinning.tsx
Comment (kamil-stasiak) [inline assets/src/pages/room/hooks/usePinning.tsx:16]: oldHistory.add() 

```suggestion
      setPinnedTileIdHistory((oldHistory) => new Set(oldHistory).add());
```

**Pull request merged**
